src/NLog.Extensions.Logging/Targets/ExtensionILoggerTarget.cs

```diff
@@ -0,0 +1,173 @@
+                                if (index-- == 0)
+                                    return new KeyValuePair<string, object>(prop.Key.ToString(), prop.Value);
+                            }
+                        }
+                        else
+                        {
+                            index -= LogEvent.Properties.Count;
+                        }
+                    }
+                    if (ContextProperties != null)
+                    {
+                        if (index < ContextProperties.Count)
+                        {
+                            foreach (var prop in ContextProperties)
+                            {
+                                if (index-- == 0)
+                                    return new KeyValuePair<string, object>(prop.Key, prop.Value);
+                            }
+                        }
+                        else
+                        {
+                            index -= ContextProperties.Count;
+                        }
+                    }
+                    if (index != 0)
+                        throw new ArgumentOutOfRangeException(nameof(index));
+                    return CreateOriginalFormatProperty();
+                }
+            }
+
+            private KeyValuePair<string, object> CreateOriginalFormatProperty()
+            {
+                return new KeyValuePair<string, object>(NLogLogger.OriginalFormatPropertyName, LogEvent.Message);
+            }
+
+            public readonly LogEventInfo LogEvent;
+            public readonly string LayoutMessage;
+            public readonly IDictionary<string, object> ContextProperties;
+
+            public LogState(LogEventInfo logEvent, string layoutMessage, IDictionary<string, object> contextProperties)
+            {
+                LogEvent = logEvent;
+                LayoutMessage = layoutMessage;
+                ContextProperties = contextProperties;
+            }
+
+            public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
+            {
+                IList<KeyValuePair<string, object>> originalMessage = new[] { CreateOriginalFormatProperty() };
+                IEnumerable<KeyValuePair<string, object>> allProperties = ContextProperties?.Concat(originalMessage) ?? originalMessage;
+                if (LogEvent.HasProperties)
+                {
+                    allProperties = LogEvent.Properties.Select(prop => new KeyValuePair<string, object>(prop.Key.ToString(), prop.Value)).Concat(allProperties);
+                }
+                return allProperties.GetEnumerator();
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return GetEnumerator();
+            }
+        }
+
+        static string LogStateFormatter(LogState logState, Exception _)
+        {
+            return logState.LayoutMessage;
+        }
+
+        Microsoft.Extensions.Logging.LogLevel LookupLogLevel(NLog.LogLevel logLevel)
```

Comment:
I prefer: ConvertToLogLevel (because: naming consistently in NLog)

Reply:
and could be static?

Reply:
Done